lib/chef/resource/chef_client_cron.rb

```diff
@@ -0,0 +1,181 @@
+
+      property :comment, String,
+        description: "A comment to place in the cron.d file."
+
+      property :config_directory, String,
+        default: Chef::Dist::CONF_DIR,
+        description: "The path of the config directory."
+
+      property :log_directory, String,
+        default: lazy { platform?("mac_os_x") ? "/Library/Logs/#{Chef::Dist::DIR_SUFFIX.capitalize}" : "/var/log/#{Chef::Dist::DIR_SUFFIX}" },
+        description: "The path of the directory to create the log file in."
+
+      property :log_file_name, String,
+        default: "client.log",
+        description: "The name of the log file to use."
+
+      property :append_log_file, [true, false],
+        default: false,
+        description: "Append to the log file instead of creating a new file on each run."
+
+      property :chef_binary_path, String,
+        default: "/opt/#{Chef::Dist::DIR_SUFFIX}/bin/#{Chef::Dist::CLIENT}",
+        description: "The path to the #{Chef::Dist::CLIENT} binary."
+
+      property :daemon_options, Array,
+        default: lazy { [] },
+        description: "An array of options to pass to the #{Chef::Dist::CLIENT} command."
+
+      action :add do
+        # TODO: Replace this with a :create_if_missing action on directory when that exists
+        unless ::Dir.exist?(new_resource.log_directory)
+          directory new_resource.log_directory do
+            owner new_resource.user
+            mode "0640"
+            recursive true
+          end
+        end
+
+        cron_d new_resource.job_name do
+          minute      new_resource.minute
+          hour        new_resource.hour
+          day         new_resource.day
+          weekday     new_resource.weekday
+          month       new_resource.month
+          environment new_resource.environment
+          mailto      new_resource.mailto if new_resource.mailto
+          user        new_resource.user
+          comment     new_resource.comment if new_resource.comment
+          command     cron_command
+        end
+      end
+
+      action :remove do
+        cron_d new_resource.job_name do
+          action :delete
+        end
+      end
+
+      action_class do
+        # Generate a uniformly distributed unique number to sleep.
+        def splay_sleep_time(splay)
+          seed = node["shard_seed"] || Digest::MD5.hexdigest(node.name).to_s.hex
+          random = Random.new(seed.to_i)
+          random.rand(splay)
+        end
+
+        def cron_command
+          cmd = ""
+          cmd << "/bin/sleep #{splay_sleep_time(new_resource.splay)}; "
+          cmd << "#{new_resource.chef_binary_path} "
+          cmd << "#{new_resource.daemon_options.join(" ")} " unless new_resource.daemon_options.empty?
+          cmd << "#{new_resource.append_log_file ? ">>" : ">"} #{::File.join(new_resource.log_directory, new_resource.log_file_name)} 2>&1"
```

Comment:
directly catting to the file seems like something we should discourage here in favor of the -L option to the client?

Reply:
Pretty sure that was just to support appending vs. overwriting

Reply:
I updated appending to use -L